ipydeps/__init__.py

```diff
@@ -290,23 +290,40 @@ def _find_overrides(packages, dep_link):
 
     return overrides
 
-def _run_get_stderr(cmd):
+def _run_get_stderr(cmd, progress=False):
     returncode = 0
     err = None
-
-    try:
-        subprocess.check_output(cmd, stderr=subprocess.PIPE)
-    except subprocess.CalledProcessError as e:
-        returncode = e.returncode
-
-        if sys.version_info.major == 3:
-            err = _bytes_to_str(e.stderr)
-        elif sys.version_info.major == 2:
-            err = e.output
+    
+    packages = cmd[5:]
+    cmd = cmd[:5]
+    
+    invalid_packages = {}
+    
+    if progress:
+        import tqdm
+        if _in_ipython():
+            progress_bar = tqdm.tqdm_notebook(total=len(packages), desc="Installing...")
         else:
-            raise Exception('Invalid version of Python')
-
-    return returncode, err
+            progress_bar = tqdm.tqdm(total=len(packages), desc="Installing...")
+            
+    for package in packages:
```

Comment:
My biggest concern with the iterative approach is that pip does some things behind the scenes when determining the order to install a list of packages:
https://github.com/pypa/pip/blob/master/src/pip/_internal/resolve.py#L367
called from
https://github.com/pypa/pip/blob/master/src/pip/_internal/commands/install.py#L366

Granted, we may already be undermining this with overrides from dependencies.json.